### PR TITLE
client exit with unrecognized arguments

### PR DESCRIPTION
**Pull request by ucasfl**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

related to [issue 9797](https://github.com/ClickHouse/ClickHouse/issues/9797)

Detailed description / Documentation draft:

...

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

### CODE REVIEW

**Review by alexey-milovidov — COMMENTED** [review]
Does it support external tables (see https://clickhouse.tech/docs/v19.14/en/operations/table_engines/external_data/)?

Also it's worth adding a simple functional test (.sh type).

**Comment by ucasFL** [comment]
it looks like we have nothing need to do about external tables.

**Comment by alexey-milovidov** [comment]
Two functional tests still has to be updated.

> Integration tests

This is Ok (Broken in #11903)

**Review by alexey-milovidov — APPROVED** [review]

File: tests/queries/0_stateless/00754_distributed_optimize_skip_select_on_unused_shards.reference [hunk 1]
```diff
@@ -1,16 +1,16 @@
 OK
 OK
 1
-OK
+FAIL
```
Comment (rschu1ze) [inline tests/queries/0_stateless/00754_distributed_optimize_skip_select_on_unused_shards.reference:4]: ~~Garbage~~ suboptimal change, corrected with #54249.

**Pull request merged**